lizrd/support/misc.py

```diff
@@ -224,3 +229,21 @@ def get_ith_chunk(tensor, chunks, i):
 
     list_of_chunks = torch.chunk(tensor, chunks, dim=0)
     return list_of_chunks[i]
+
+
+def calculate_current_batch_size_from_rampup(
+    processed_tokens,
+    transition_points,
+    batch_sizes,
+    target_batch_size,
+):
+    transition_points_in_tokens = [point * 1e9 for point in transition_points]
+
+    if processed_tokens >= transition_points_in_tokens[-1]:
+        return target_batch_size
+
+    for i in reversed(range(len(transition_points_in_tokens) - 1)):
+        if processed_tokens >= transition_points_in_tokens[i]:
+            return batch_sizes[i + 1]
+
+    return batch_sizes[0]
```

Comment:
Just a suggestion. Suppose our object BatchSizeRampupConfig instead is a list of pairs (transition_point, batch_size). We can simplify a function : 
```suggestion
def calculate_current_batch_size_from_rampup(
    processed_tokens,
    rampup_transition_config: List[int, int], 
    target_batch_size,
):
    for transition_point, batch_size in rampup_transition_config:
        if processed_tokens < transition_point * 1e9:
            return batch_size
    return target_batch_size
```

Reply:
We can also add 
```math
f(processedtokens) = 
     \begin{cases}
       \text{16} &\quad\text{if} 0 \leq \text{processedtokens} \le 0.5 \\
       \text{64} &\quad\text{if} 0.5 \leq \text{processedtokens} \le 1 \\
       \text{128} &\quad\text{if} 1 \leq \text{processedtokens} \le 2 \\
       \text{512} &\quad \text{otherwise}
     \end{cases}
```

Reply:
Right! I will change it as suggested by you